api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java

```diff
@@ -146,19 +146,27 @@ public ClassLoader run() {
         ConfigProviderResolver instance = loadSpi(parentcl);
 
         if (instance == null) {
-            ServiceLoader<ConfigProviderResolver> sl = ServiceLoader.load(
-                            ConfigProviderResolver.class, cl);
-            for (ConfigProviderResolver spi : sl) {
-                if (instance != null) {
-                    throw new IllegalStateException(
-                                    "Multiple ConfigResolverProvider implementations found: "
-                                                    + spi.getClass().getName() + " and "
-                                                    + instance.getClass().getName());
-                }
-                else {
-                    instance = spi;
+            Thread thread = Thread.currentThread();
+            ClassLoader original = thread.getContextClassLoader();
+
+            try {
+                thread.setContextClassLoader(cl);
+
+                for (ConfigProviderResolver spi : ServiceLoader.load(ConfigProviderResolver.class)) {
```

Comment:
ServiceLoader also has a load() method which takes the ClassLoader. That would avoid having to set the TCCL, isn't?

Reply:
The Service Loader Mediator spec only supports the `ServiceLoader.load(Class<?>)` method. That's why I changed this logic, though it's exactly the same because this method uses the TCCL first. So the logic of first setting the TCCL and then using load(Class<?>) is exactly the same as using `load(Class<?>, ClassLoader)`.

Reply:
Not all OSGi environment supports Service Loader Mediator.

Reply:
all modern do. And for the rest it's rather easy to provide an own ConfigProviderResolver.

Reply:
Gonna resolve this conversation since in the end we didn't need to change this code, we fixed the SpiFly impl to support this ServiceLoader method.